packages/salesforcedx-vscode-core/src/orgBrowser/orgMetadata.ts

```diff
@@ -0,0 +1,145 @@
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {
+  CliCommandExecutor,
+  Command,
+  SfdxCommandBuilder
+} from '@salesforce/salesforcedx-utils-vscode/out/src/cli';
+import {
+  ContinueResponse
+} from '@salesforce/salesforcedx-utils-vscode/out/src/types';
+import * as fs from 'fs';
+import * as path from 'path';
+import { Observable } from 'rxjs/Observable';
+import { isNullOrUndefined } from 'util';
+import * as vscode from 'vscode';
+import { channelService } from '../channels';
+import {
+  EmptyParametersGatherer,
+  SfdxCommandlet,
+  SfdxCommandletExecutor,
+  SfdxWorkspaceChecker
+} from '../commands';
+import { nls } from '../messages';
+import { notificationService, ProgressNotification } from '../notifications';
+import { taskViewService } from '../statuses';
+import { getRootWorkspacePath, hasRootWorkspace, OrgAuthInfo } from '../util';
+
+export class ForceDescribeMetadataExecutor extends SfdxCommandletExecutor<string> {
+  private outputPath: string;
+
+  public constructor(outputPath: string) {
+    super();
+    this.outputPath = outputPath;
+  }
+
+  public build(data: {}): Command {
+    return new SfdxCommandBuilder()
+      .withDescription(
+       'SFDX: Describe Metadata'
+      )
+      .withArg('force:mdapi:describemetadata')
+      .withJson()
+      .withFlag('-f', this.outputPath)
+      .withLogName('force_describe_metadata')
+      .build();
+  }
+
+  public execute(response: ContinueResponse<string>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      buildTypesList(this.outputPath);
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceDescribeMetadata(outputPath?: string) {
+  if (!isNullOrUndefined(outputPath)) {
+    const describeExecutor = new ForceDescribeMetadataExecutor(outputPath);
+    const commandlet = new SfdxCommandlet(
+      workspaceChecker,
+      parameterGatherer,
+      describeExecutor
+    );
+    await commandlet.run();
+  }
+}
+
+async function getMetadataTypesPath(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
```

Comment:
Since you are already checking for `hasRootWorkspace()` here, could you call `OrgAuthInfo.getDefaultUsernameOrAlias()` directly fro line 94 and get rid of the helper function in line 109?

Reply:
Yeah, I think that would definitely make sense!